cadence/gateway/cadence/interface.py

```diff
@@ -0,0 +1,11 @@
+from abc import abstractmethod
+from typing import Tuple
+
+from cadence.cadence_types import ListDomainsRequest, ListDomainsResponse
+
+
+class CadenceGatewayInterface:
```

Comment:
(please ignore my previous comment).
How about `CadenceServiceInterface`? And having the other two as `CadenceTChannelService` and `CadenceGrpcService`. 
I think gateway is more used as server.

Reply:
Please feel free to add as many comments like this one as possible. My experience is mainly with servers so I might bring some "bad practices". Fixed